spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -119,6 +120,24 @@ public Blob getGoogleStorageObject() throws IOException {
 		return this.storage.get(getBlobId());
 	}
 
+	/**
+	 * Creates a signed URL to an object if it exists. This method will fail if this storage resource
+	 * was not created using service account credentials.
+	 * @param timeUnit the time unit used to determine how long the URL is valid.
+	 * @param timePeriods the number of periods to determine how long the URL is valid.
+	 * @return the URL if the object exists, and null if it does not.
+	 * @throws IOException
+	 */
+	public URL createSignedUrl(TimeUnit timeUnit, long timePeriods) throws IOException {
+		Blob blob = this.getGoogleStorageObject();
+		if (blob == null) {
+			return null;
+		}
+		return this.storage.signUrl(
```

Comment:
Have you tested this change without using application default credentials?

From reading the `Storage#signUrl()` documentation, I think we do need to pass it a `SignUrlOptions` object containing the desired service account credentials.
In other words, I don't think the `Storage` object is smart enough to use the `GoogleCredentials` that it was passed in its constructor, but I'd be curious to test if this is true.

Reply:
I did try it just now and it did make a working URL for me. This part of the doc makes it sound like it is aware of the credentials with which it was made:

 If the credentials passed to
   * {@link StorageOptions} do not implement {@link ServiceAccountSigner} (this is the case, for
   * instance, for Compute Engine credentials and Google Cloud SDK credentials) then {@code signUrl}
   * will throw an {@link IllegalStateException} unless an implementation of
   * {@link ServiceAccountSigner} is passed using the
   * {@link SignUrlOption#signWith(ServiceAccountSigner)} option.

Reply:
Beautiful!